xWorkflows/conslidated_xworkflows/README.md

```diff
@@ -0,0 +1,175 @@
+
+
+
+## Overview
+
+The Consolidated xWorkflow simplifies the process of defining and managing sworkflows by allowing users to specify them in a straightforward YAML configuration file. The **query_pod** application, which is central to this system, interfaces with various Kubecost APIs to fetch relevant data. This data is then  stored in MongoDB according to the workflows outlined in the configuration file. This setup reduces the need for technical expertise and streamlines the xworkflow configuration process for end user.
+
+## Kubernetes Configuration
+
+The Kubernetes configuration includes several components for deploying the query pod and MongoDB. These configurations are located in the `k8s_yaml_files` directory.
+
+### Kubernetes YAML Files
+
+1. **`mongodb-headless-service.yaml`**: Defines a headless service for MongoDB to enable stable network identities and facilitate StatefulSet communication.
+2. **`mongodb-statefulset.yaml`**: Configures the MongoDB StatefulSet, including initialization parameters and storage.
+3. **`query-pod-deployment.yaml`**: Specifies the deployment for the query pod, including environment variables, image details, and volume mounts.
+4. **`workflow-configmap.yaml`**: Contains the configuration for the workflow definitions used by the query pod.
+5. **`mongodb-secrets.yaml`**: Stores MongoDB credentials securely.
+
+## Python Application (query_pod)
+
+### Description
+
+The `query_pod` application is a Python script (`app.py`) designed to:
+
+1. **Read Workflow Definitions**: Load and parse a YAML file containing workflow configurations.
+2. **Fetch Data**: Retrieve data from specified kubecost APIs based on workflow parameters.
+3. **Store Data**: Insert the fetched data into MongoDB collections.
+
+### Functionality
+
+1. **Connecting to MongoDB**: Establishes a connection to the MongoDB instance using credentials provided as environment variables.
+2. **Fetching Data**: Retrieves data from kubecost APIs using predefined URLs and parameters.
+3. **Processing Workflows**: Iterates over workflows, sets up the scraper, and stores data accordingly.
+
+## How to Setup
+
+### Build Docker Image
+
+1. Navigate to the `query_pod_codebase` directory where the `Dockerfile` is located.
+2. Build the Docker image using the following command:
+   ```sh
+   docker build -t your-image-name:tag .
+   ```
+
+### Update Kubernetes YAML Files
+
+1. Update the `query-pod-deployment.yaml` file with the new image name and tag.
+2. Apply the updated YAML file to your Kubernetes cluster:
+   ```sh
+   kubectl apply -f k8s_yaml_files/query-pod-deployment.yaml
+   ```
+
+Here’s an enhanced section on how to add a new workflow to the README:
+
+---
+
+### Adding a New Workflow
+
+To add a new workflow to the **query_pod** application, follow these steps:
+
+#### 1. Define Workflow in `values.yaml`
+
+Add the new workflow configuration to the `values.yaml` file in the `workflow-configmap.yaml` ConfigMap. For example:
```

Comment:
add configuration to values,yaml file 'in' workflow-configmap.yaml ConfigMap does not make sense.